cli_helpers/tabular_output/terminaltables_adapter.py

```diff
@@ -25,4 +24,10 @@ def adapter(data, headers, table_format=None, **kwargs):
     table = table_format_handler[table_format]
 
     t = table([headers] + list(data), **filter_dict_by_key(kwargs, keys))
-    return t.table
+
+    dimensions = terminaltables.width_and_alignment.max_dimensions(
+        t.table_data,
+        t.padding_left,
+        t.padding_right)[:3]
+    for row in t.gen_table(*dimensions):
+        yield ''.join(''.join(row))
```

Comment:
I think they are using two joins because they join the pieces of the row together, then they join all the rows together.

We only need to join the pieces of the row together. If you put logging in you should see that the first join is returning a string, not a list. At least that's what I was seeing yesterday when I was testing it.